test/lib/ufe/testAttributeBlock.py

```diff
@@ -261,5 +261,91 @@ def testAttributeBlocking3dCommonApi(self):
+        rootLayer = stage.GetRootLayer()
+        subLayerC = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerC")[0]
+        subLayerB = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerB")[0]
+        subLayerA = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerA")[0]
+
+        # check to see the sublayers added
+        addedLayers = [subLayerA, subLayerB, subLayerC]
+        self.assertEqual(rootLayer.subLayerPaths, addedLayers)
+
+        # set the edit target to LayerB
+        cmds.mayaUsdEditTarget(proxyShapePath, edit=True, editTarget=subLayerB)
+
+        # create a xformable and give it a matrix4d xformOp:transform
+        # This will match the UsdTransform3dMatrixOp, but not the Maya API.
+        cylinderXformable = UsdGeom.Xformable(cylinderPrim)
+        transformOp = cylinderXformable.AddTransformOp()
+        xform = Gf.Matrix4d(1.0).SetTranslate(Gf.Vec3d(10, 20, 30))
+        transformOp.Set(xform)
+        self.assertTrue(cylinderPrim.GetPrim().HasAttribute("xformOp:transform"))
+
+        # Now that we have set up the transform stack to be 3dMatrixOp compatible,
+        # create the Transform3d interface.
+        cylinderPath = ufe.Path([
+                     mayaUtils.createUfePathSegment("|mayaUsdTransform|shape"), 
+                     usdUtils.createUfePathSegment("/pCylinder1")])
+        cylinderItem = ufe.Hierarchy.createItem(cylinderPath)
+
+        cylinderT3d = ufe.Transform3d.transform3d(cylinderItem)
+
+        # select the cylinderItem
+        sn = ufe.GlobalSelection.get()
+        sn.clear()
+        sn.append(cylinderItem)
+
+        # move
+        cmds.move(65, 20, 10)
+        self.assertEqual(cylinderXformable.GetXformOpOrderAttr().Get(),
+                    Vt.TokenArray(('xformOp:transform', )))
+
+        # validate the Matrix4d entires
+        actual = cylinderXformable.GetLocalTransformation()
+        expected  = Gf.Matrix4d( 1, 0, 0, 0,
+                                 0, 1, 0, 0,
+                                 0, 0, 1, 0,
+                                 65, 20, 10, 1)
+
+        self.assertTrue(Gf.IsClose(actual, expected, 1e-5))
+
+        # set the edit target to a weaker layer (LayerC)
+        cmds.mayaUsdEditTarget(proxyShapePath, edit=True, editTarget=subLayerC)
+
+        # check if transform attribute is editable
+        transformAttr = cylinderPrim.GetAttribute('xformOp:transform')
+        self.assertIsNotNone(transformAttr)
+
+        # authoring new transformation edit is not allowed.
+        self.assertFalse(mayaUsdUfe.isAttributeEditAllowed(transformAttr))
+
+        # set the edit target to a stronger layer (LayerA)
+        cmds.mayaUsdEditTarget(proxyShapePath, edit=True, editTarget=subLayerA)
+
+        # authoring new transformation edit is allowed.
+        self.assertTrue(mayaUsdUfe.isAttributeEditAllowed(transformAttr))
```

Comment:
Line 335 to 348: tests blocking attribute authoring in weaker or stronger layers for the `xformOp:transform`.